app/backend/src/User/controller/UserController.ts

```diff
@@ -0,0 +1,134 @@
+    service: UserService,
+  ) {
+    super(service);
+    this.authService = new AuthService(userModel);
+  }
+
+  create = async (
+    req: RequestWithBody<User>,
+    res: Response<User | ResponseError | object>,
+  ): Promise<typeof res> => {
+    const { body } = req;
+
+    try {
+      const user = await this.service.create(body);
+      const { name, email } = body;
+
+      if(!user) {
+        return res.status(500).json({ error: this.errors.internal});
+      }
+
+      if('error' in user) {
+        return res.status(400).json({ error: user.error });
+      }
+
+      const token = this.authService.genToken(user);
+
+      return res.status(201).json({ name, email, token});
+    } catch (err) {
+      return res.status(500).json({ error: this.errors.internal });
+    }
+  };
+
+  read = async (
+    req: RequestWithBody<User>,
+    res: Response<User[] | ResponseError>,
+  ): Promise<typeof res> => {
+    try {
+      const users = await this.service.read();
+
+      if(!users) {
+        return res.status(404).json({error: this.errors.notFound });
+      }
+
+      return res.status(200).json(users);
+    } catch (err) {
+      return res.status(500).json({error: this.errors.internal });
+    }
+  };
+
+  delete = async (
+    req: RequestWithBody<User>,
+    res: Response<User | ResponseError>): Promise<typeof res> => {
+    try {
+      const { id } = req.params;
+      const user = await this.service.delete(id);
+
+      if(!user) {res.status(404).json({ error: this.errors.notFound });}
+
+      return res.status(204).end();
+    } catch (err) {
+      return res.status(500).json({ error: this.errors.internal });
+    }
+  };
+
+  findOneAndDelete = async (
+    req: RequestWithBody<User>,
+    res: Response<User | ResponseError>
+  ): Promise<typeof res> => {
+    try {
+      const user = await this.service.findOneAndDelete(req.body);
+
+      if(!user) {res.status(404).json({ error: this.errors.notFound });}
+
+      return res.status(200).end();
+    } catch (error) {
+      return res.status(500).json({ error: this.errors.internal });
+    }
+  };
+
+  login = async(
+    req: RequestWithBody<User>,
+    res: Response<User | ResponseError| string | object>
+  ): Promise<typeof res> => {
+    //authenticate user
+
+    const response = await this.authService.authorization(req.body);
+    return res.status(200).json({response});
+  };
+
+  auth = (
+    req: RequestWithBody<User>,
+    res: Response<object | ResponseError>,
+    next: NextFunction,
+  ): typeof res | void =>{
+
+    const token = req.headers.authorization;
+
+    const response = this.authService.authentication(token);
+
+    if(response.code !== 200) {
+      res.status(response.code).json({ message: response.data });
+    }
+    next();
+  };
+
+  update = async (
+    req: RequestWithBody<User & { id: string }>,
+    res: Response<User | ResponseError>,
+  ): Promise<typeof res>=> {
+    const { body } = req;
+    const { id } = req.params;
+
+    try {
```

Comment:
this looks like a nested try catch like we talked before, the mongoose ORM will handle the problem and throw an error in case of issues, and later the `response?` will also do this so its redundant 

we could try debug to invoke the error and see if mongoose sends us the proper message or treats it and a generic one, in case if its generic we can keep for now but it will require some refactors